cli_tools/google-osconfig-agent/patch/watcher.go

```diff
@@ -88,9 +94,30 @@ func watcher(ctx context.Context) {
 			time.Sleep(5 * time.Second)
 			continue
 		}
-		id := strings.Split(md, ",")[0]
-		if id != "" {
-			ackPatch(ctx, id)
+
+		patchNotification, err := metadata.InstanceAttributeValue(metadataPatchKey)
+		if err != nil {
+			switch err.(type) {
+			case metadata.NotDefinedError:
```

Comment:
why is notdefinederror a special case? just for logging?  is it the case that the metadata key is eventually going to be available, because if it is not, then does it make sense to retry to fetch the attribuevalue?

Reply:
i see... yeah, my initial impression was why can't the block be written as
`if err != nil {
    logger.Errorf("error line...")
    conitnue
}`
but now i understand, we explicitly want to call out that keynotpresent is not really an error for our application. totally makes sense.
thanks for the explanation.